luarules/gadgets/game_tax_resource_sharing.lua

```diff
@@ -28,22 +28,35 @@ local spGetTeamUnitCount = Spring.GetTeamUnitCount
 local gameMaxUnits = math.min(Spring.GetModOptions().maxunits, math.floor(32000 / #Spring.GetTeamList()))
 
 local sharingTax = Spring.GetModOptions().tax_resource_sharing_amount
+local metalTaxThreshold = Spring.GetModOptions().player_metal_send_threshold or 0 -- Use standardized key
+
+-- Table to store cumulative metal sent: cumulativeMetalSent[senderTeamId]
+local cumulativeMetalSent = {}
 
 ----------------------------------------------------------------
--- Callins
+-- Initialization
 ----------------------------------------------------------------
 
+function gadget:Initialize()
+	-- Initialize cumulative tracking for all potential senders
+	local teamList = Spring.GetTeamList()
+	for _, senderID in ipairs(teamList) do
+		cumulativeMetalSent[senderID] = 0 -- Initialize per sender
+	end
+end
 
+----------------------------------------------------------------
+-- Callins
+----------------------------------------------------------------
 
 function gadget:AllowResourceTransfer(senderTeamId, receiverTeamId, resourceType, amount)
-
 	-- Spring uses 'm' and 'e' instead of the full names that we need, so we need to convert the resourceType
 	-- We also check for 'metal' or 'energy' incase Spring decides to use those in a later version
-	local resourceName
-	if (resourceType == 'm') or (resourceType == 'metal') then
-		resourceName = 'metal'
-	elseif (resourceType == 'e') or (resourceType == 'energy') then
-		resourceName = 'energy'
+	local resourceName -- This variable will hold the standardized name
+	if (resourceType == 'm') or (resourceType == 'metal') then -- Checks for 'm' OR 'metal'
+		resourceName = 'metal' -- Standardizes to 'metal'
+	elseif (resourceType == 'e') or (resourceType == 'energy') then -- Checks for 'e' OR 'energy'
+		resourceName = 'energy' -- Standardizes to 'energy'
```

Comment:
i'm in no position to review this since im not a contributor, but i think comments like this do more harm than good. 

https://gist.github.com/wojteklu/73c6914cc446146b8b533c0988cf8d29#comments-rules 

These comments "violate" points 2 and 3: "Don't be redundant.
Don't add obvious noise."

Reply:
I totally agree I'll remove them.